### PR TITLE
Update README to Fastify v4

### LINKED ISSUE #215: Example in read me does not work at all
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.3.0

### Plugin version

8.0.0

### Node.js version

18.7.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.4

### Description

The plugin does literally nothing. The response is not decorated and the plugin seems to have not even run.

### Steps to Reproduce

```js
import Fastify from 'fastify';
import cors from '@fastify/cors';
import middie from '@fastify/middie';

const fastify = Fastify();

await fastify.register(middie);
await fastify.register(cors);
```

I also tried [the example](https://www.fastify.io/docs/latest/Reference/Middleware/) from fastify.io, and that crashes immediately:

```js
await fastify.register(middie);
fastify.use(cors());
```

```console
[…]/@fastify/cors/index.js:20
  fastify.decorateRequest('corsPreflightEnabled', false)
          ^
TypeError: Cannot read properties of undefined (reading 'decorateRequest')
```

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #215

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,17 +21,19 @@ npm i @fastify/cors
...
+await fastify.register(cors, { 
   // put your options here
 })
 
 fastify.get('/', (req, reply) => {
   reply.send({ hello: 'world' })
 })
 
-fastify.listen(3000)
+await fastify.listen(3000)
```
Comment (Uzlopak) [inline README.md:36]: isnt this notation not deprecated?

```suggestion
await fastify.listen({ port: 3000 })
```

File: README.md [hunk 2]
```diff
@@ -21,17 +21,19 @@ npm i @fastify/cors
 ## Usage
 Require `@fastify/cors` and register it as any other plugin, it will add a `preHandler` hook and a [wildcard options route](https://github.com/fastify/fastify/issues/326#issuecomment-411360862).
 ```js
-const fastify = require('fastify')()
+import Fastify from 'fastify'
```
Comment (Uzlopak) [inline README.md:24]: I would prefer the require notation.

**Review by Uzlopak — CHANGES_REQUESTED** [review]

**Review by Uzlopak — APPROVED** [review]
LGTM

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Review by Eomm — APPROVED** [review]

**Pull request merged**